lib/logstash/filters/cidr.rb

```diff
@@ -35,11 +35,88 @@ class LogStash::Filters::CIDR < LogStash::Filters::Base
   #     }
   config :network, :validate => :array, :default => []
 
+  # The full path of the external file containing the IP network(s) to check against. Example:
+  # [source,ruby]
+  #     filter {
+  #       %PLUGIN% {
+  #         add_tag => [ "linklocal" ]
+  #         address => [ "%{clientip}" ]
+  #         network_path => "/etc/logstash/networks"
+  #       }
+  #     }
+  # NOTE: it is an error to specify both 'network' and 'network_path'.
+  config :network_path, :validate => :path
+
+  # The separator character used in the encoding of the external file pointed by network_path.
+  config :separator,:validate => :string, :default => "\n"
+
+  # When using a network list from a file, this setting will indicate how frequently
+  # (in seconds) logstash will check the file for updates.
+  config :refresh_interval, :validate => :number, :default => 600
+
   public
-  def register
-    # Nothing
+  def register    
+    rw_lock = java.util.concurrent.locks.ReentrantReadWriteLock.new
+    @read_lock = rw_lock.readLock
+    @write_lock = rw_lock.writeLock
+
+    if @network_path && !@network.empty? #checks if both network and network path are defined in configuration options
+    	raise LogStash::ConfigurationError, I18n.t(
+    	"logstash.agent.configuration.invalid_plugin_register",
+    	:plugin => "filter",
+    	:type => "cidr",
+    	:error => "The configuration options 'network' and 'network_path' are mutually exclusive"
+    	)
```

Comment:
Sorry to be a pest about this, but the lines in this block are _still_ using tabs. Please do not use tabs for indentation. Also, the nested continuation lines should be indented an extra level:

```ruby
raise LogStash::ConfigurationError, I18n.t(
  "logstash.agent.configuration.invalid_plugin_register",
  :plugin => "filter",
  :type => "cidr",
  :error => "The configuration options 'network' and 'network_path' are mutually exclusive"
)
```

There are a few more tabs further down in the file, so please scan the whole file for tabs.